### PR TITLE
[DEV-1951] Add urlreplacemap codec, + tests + mapping function

### PR DESCRIPTION
**Pull request by tommaso1**
#### List of Changes

Adds the codec to read and parse urlreplacemap from strapi
Depends on [[DEV-1950]](https://github.com/pagopa/developer-portal/pull/1161)
#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

[DEV-1950]: https://pagopa.atlassian.net/browse/DEV-1950?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/healthy-cobras-type.md
Comment (MarBert) [inline .changeset/healthy-cobras-type.md:1]: This is the changeset from DEV-1950. Shouldn't there be a changeset for DEV-1951 ?

**Review by MarBert — DISMISSED** [review]

File: apps/nextjs-website/src/lib/strapi/__tests__/urlReplaceMap.test.ts [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                      size: 1207.47,
+                      url: 'http://localhost:1337/uploads/download_d1d23bf79f.jpeg',
+                      previewUrl: null,
+                      provider: 'strapi-provider-upload-custom',
+                      provider_metadata: null,
+                      createdAt: '2024-09-24T16:19:29.152Z',
+                      updatedAt: '2024-09-24T16:19:29.152Z',
+                    },
+                  },
+                },
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/__tests__/urlReplaceMap.test.ts:103]:
```suggestion
                image: mediaRasterJson,
```

File: apps/nextjs-website/src/lib/strapi/__tests__/urlReplaceMap.test.ts [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/__tests__/urlReplaceMap.test.ts:175]:
```suggestion
                mobileImage: mediaRasterJson,
```

File: .changeset/healthy-cobras-type.md
Comment (marcobottaro) [reply .changeset/healthy-cobras-type.md:1]: After the merge of the pull request DEV-1950 this file with all the changes in `strapi-cms/src` should desappear

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/strapi/codecs/UrlReplaceMapCodec.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import * as t from 'io-ts/lib';
+import { GuideCodec } from './GuidesCodec';
+import { NullToUndefinedCodec } from './NullToUndefinedCodec';
+
+const UrlToGuideCodec = t.strict({
+  id: t.number,
+  url: t.string,
+  version: t.union([NullToUndefinedCodec, t.string]),
+  guide: t.strict({
+    data: GuideCodec,
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/codecs/UrlReplaceMapCodec.ts:10]:
```suggestion
    data: t.union([NullToUndefinedCodec, GuideCodec]),
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Pull request merged**